From: "Neal H. Walfield" <neal@gnu.org>
Date: Fri, 5 Jun 2015 05:29:03 +0200
Subject: secmem: When clearing memory,
 don't clear beyond the end of the buffer.

* secmem/secmem.c (secmem_malloc): Only clear the user memory; don't
clear beyond the end of the buffer.
---
 secmem/secmem.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/secmem/secmem.c b/secmem/secmem.c
index 9a478cf..b4dc5a2 100644
--- a/secmem/secmem.c
+++ b/secmem/secmem.c
@@ -363,7 +363,8 @@ secmem_malloc( size_t size )
     if( cur_blocks > max_blocks )
 	max_blocks = cur_blocks;
 
-    memset (&mb->u.aligned.c, 0, size);
+    memset (&mb->u.aligned.c, 0,
+	    size - (size_t) &((struct memblock_struct *) 0)->u.aligned);
 
     return &mb->u.aligned.c;
 }
